jiant/utils/utils.py

```diff
@@ -323,22 +325,48 @@ def select_task_specific_args(exp_args, diff_args):
     return diff_args
 
 
-def load_model_state(model, state_path, gpu_id, skip_task_models=[], strict=True):
+def get_state_dict_for_loading(model, model_state) -> nn.Module:
+    """ Function for making sure state dict keys are named appropriately for 
+        multi-GPU and single-GPU use cases
+
+    Parameters
+    ----------
+    model: The model object to populate with loaded parameters.
+    model_state: collections.OrderdDict of model state
+    """
+    final_model_state = collections.OrderedDict()
+
+    def get_key(name):
+        key = name
+        if isinstance(model, nn.DataParallel):
+            if "module" not in key:
+                key = "module.%s" % key
+        else:
+            if "module" in key:
+                key = key.replace("module.", "")
```

Comment:
Make this check explicitly for prefix with `.startswith`, and drop the first n characters (in case module appears somewhere else in the parameter name).